### PR TITLE
Bug 1695494 - Adding resource checks to all components

### PR DESCRIPTION
**Pull request by ewolinetz**
Adding checks for the different logging components to check if their resources have changed, warranting a rollout.

Ref: https://bugzilla.redhat.com/show_bug.cgi?id=1695494

### CODE REVIEW

File: pkg/k8shandler/collection.go [hunk 1]
```diff
@@ -323,6 +329,25 @@ func updateCurrentDaemonsetImages(current *apps.DaemonSet, desired *apps.DaemonS
 	return current
 }
 
+func areResourcesDifferent(current *apps.Daemonset, desired *apps.Daemonset) (bool, *apps.Daemonset) {
```
Comment (richm) [inline pkg/k8shandler/collection.go]: you could have a single `areResourcesDifferent` function - use introspection to determine if object is a Daemonset or a Deployment or a CronJob or whatever

File: pkg/k8shandler/curation.go [hunk 2]
```diff
@@ -315,5 +315,11 @@ func isCuratorDifferent(current *batch.CronJob, desired *batch.CronJob) (*batch.
 		different = true
 	}
 
+	if changed, updated := utils.CompareResources(current.Spec.JobTemplate.Spec.Template.Spec.Containers[0], desired.Spec.JobTemplate.Spec.Template.Spec.Containers[0]); changed {
```
Comment (richm) [inline pkg/k8shandler/curation.go]: use `areResourcesDifferent`

**Comment by richm** [comment]
ok - if it passes, squash it and i'll lgtm

**Comment by ewolinetz** [comment]
@richm should pass now, confirmed it works on my local env. lets make sure ci tests come back clean first

File: pkg/k8shandler/collection.go [hunk 3]
```diff
@@ -261,6 +261,11 @@ func isDaemonsetDifferent(current *apps.DaemonSet, desired *apps.DaemonSet) (*ap
 		different = true
 	}
 
+	if utils.AreResourcesDifferent(current, desired) {
+		logrus.Infof("Daemonset resource(s) change found, updating %q", current.Name)
```
Comment (richm) [inline pkg/k8shandler/collection.go]: Use `Collector` instead of `Daemonset` to be consistent with the other log messages below

Comment (ewolinetz) [reply pkg/k8shandler/collection.go]: updated

**Pull request merged**